lib/npg_pipeline/pluggable.pm

```diff
@@ -612,6 +635,19 @@ sub _run_function {
     $attrs->{$key} = $value;
   }
 
+  ####
+  # Bring in resource requirements that functions need in order to
+  # create task definitions
+
+  # Extract default properties from graphwide metadata
+  my $jgraph = $self->_function_list_conf;
+  my $resources = $jgraph->{graph}{metadata}{default_resources};
```

Comment:
The graph-level resourcebeing the same for all functions, it can be cashed in a private attribute

Reply:
I agree in principle, but I think all this stuff is already there after the first access, unless pluggable gets instantiated multiple times. I'm happy to add but I don't think it would achieve a lot.

Reply:
OK